src/lua/api-gateway/util/OauthClient.lua

```diff
@@ -16,6 +16,52 @@ function OauthClient:new(o)
     return o
 end
 
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        require(module)
+    end
+
+    local res = pcall(requiref, module)
+    if not (res) then
+        ngx.log(ngx.WARN, "Could not load module [", module, "].")
+        return nil
+    end
+    return require(module)
+end
+
+local restyDogstatsd = loadrequire('resty_dogstatsd')
+
+--- Returns an instance of dogstatsd only if it does not already exist
+function OauthClient:getDogstatsd()
+    if self.dogstatsd ~= nil then
+        return self.dogstatsd
+    end
+
+    local dogstatsd = restyDogstatsd.new({
+        statsd = {
+            host = "datadog.docker",
+            port = 8125,
+            namespace = "api_gateway",
+        },
+        tags = {
+            "application:lua",
+        },
+    })
+    return dogstatsd
+end
+
+local oauthCalls = 'oauth.http_calls'
```

Comment:
????

Reply:
This is the string used to query datatdog metrics. I extracted it into a variable in case we need to change the name.